zk/tests/unwinds/unwind.sh

```diff
@@ -11,103 +11,179 @@
-rm -rf "$dataPath/phase1-diffs"
-rm -rf "$dataPath/phase2-diffs"  
+
+# rm -rf "$dataPath/phase1-diffs"
+# rm -rf "$dataPath/phase2-diffs"
 
 # run datastream server
+echo -e '\nStarting datastream server \n'
 go run ./zk/debug_tools/datastream-host --file="$(pwd)/zk/tests/unwinds/datastream/hermez-dynamic-integration8-datastream/data-stream.bin" &
 
-# in order to start the datastream server
-sleep 10
+dspid=$! # get the id of the DS process
+
+echo "Waiting for datastream server to become available on port 6900..."
+while ! bash -c "</dev/tcp/0.0.0.0/6900" 2>/dev/null; do
+    sleep 1
+done
+echo "Datastream server is now available."
 
-# run erigon for a while to sync to the unwind point to capture the dump
-timeout $firstTimeout ./build/bin/cdk-erigon \
+# try with 1 and check on time
+echo -e '\nRun Erigon to BlockHeight: ' "${firstStop}" '\n'
+./build/bin/cdk-erigon \
     --datadir="$dataPath/rpc-datadir" \
-    --config=./dynamic-integration8.yaml \
-    --zkevm.sync-limit=${firstStop}
+    --config="zk/tests/unwinds/config/dynamic-integration8.yaml" \
+    --debug.limit=1 \
+    --debug.limit="${firstStop}"
+
+echo -e '\nDumping data \n'
 
 # now get a dump of the datadir at this point
 go run ./cmd/hack --action=dumpAll --chaindata="$dataPath/rpc-datadir/chaindata" --output="$dataPath/phase1-dump1"
 
+echo -e '\nRun Erigon to Block Height: ' "${stopBlock}" '\n'
 # now run to the final stop block
-timeout $secondTimeout ./build/bin/cdk-erigon \
+./build/bin/cdk-erigon \
     --datadir="$dataPath/rpc-datadir" \
-    --config=./dynamic-integration8.yaml \
-    --zkevm.sync-limit=${stopBlock}
+    --config="zk/tests/unwinds/config/dynamic-integration8.yaml" \
+    --debug.limit="${stopBlock}"  
 
+echo -e '\nDumping data phase 2 \n'
 # now get a dump of the datadir at this point
 go run ./cmd/hack --action=dumpAll --chaindata="$dataPath/rpc-datadir/chaindata" --output="$dataPath/phase2-dump1"
 
 # now run the unwind
+echo -e '\nUnwinding to batch: ' "${unwindBatch}" '\n'  
 go run ./cmd/integration state_stages_zkevm \
     --datadir="$dataPath/rpc-datadir" \
-    --config=./dynamic-integration8.yaml \
+    --config="zk/tests/unwinds/config/dynamic-integration8.yaml" \
     --chain=dynamic-integration \
-    --unwind-batch-no=${unwindBatch}
+    --unwind-batch-no="${unwindBatch}"  
 
+echo -e '\nDumping data after unwind \n'
 # now get a dump of the datadir at this point
 go run ./cmd/hack --action=dumpAll --chaindata="$dataPath/rpc-datadir/chaindata" --output="$dataPath/phase1-dump2"
 
-
-mkdir -p "$dataPath/phase1-diffs/pre"
-mkdir -p "$dataPath/phase1-diffs/post"
+# mkdir -p "$dataPath/phase1-diffs/pre"
+# mkdir -p "$dataPath/phase1-diffs/post"
+
+different_files=(
+    "Code.txt"
+    "HashedCodeHash.txt"
+    "hermez_l1Sequences.txt"
+    "hermez_l1Verifications.txt"
+    "HermezSmt.txt"
+    "PlainCodeHash.txt"
+    "SyncStage.txt"
+    "BadHeaderNumber.txt"
+    "CallToIndex.txt"
+)
+
+is_in_array() {
```

Comment:
I also did this but reverted because I assumed it was better to read the other way around, but it doesn't matter.

I think the multiple OR was faster then having a FOR